pkgs/stdenv/generic/setup.sh

```diff
@@ -16,29 +16,15 @@ if (( "${NIX_DEBUG:-0}" >= 6 )); then
     set -x
 fi
 
-if [ -f .attrs.sh ]; then
+if [ -f .attrs.sh ] || [[ -n "${NIX_ATTRS_JSON_FILE:-}" ]]; then
     __structuredAttrs=1
     echo "structuredAttrs is enabled"
-else
-    __structuredAttrs=
-fi
 
-if [ -n "$__structuredAttrs" ]; then
     for outputName in "${!outputs[@]}"; do
         # ex: out=/nix/store/...
         export "$outputName=${outputs[$outputName]}"
     done
 
-    # Before Nix 2.4, $NIX_ATTRS_*_FILE was named differently:
-    # https://github.com/NixOS/nix/commit/27ce722
-    if [[ -n "${ATTRS_JSON_FILE:-}" ]]; then
-        export NIX_ATTRS_JSON_FILE="$ATTRS_JSON_FILE"
-    fi
-
-    if [[ -n "${ATTRS_SH_FILE:-}" ]]; then
-        export NIX_ATTRS_SH_FILE="$ATTRS_SH_FILE"
-    fi
```

Comment:
Was added in https://github.com/nixos/nixpkgs/commit/1a29857b8a93f5259f0c2e919becc0bf9db24f85

@sternenseemann

Reply:
> Nix 2.3 is the minimum Nix supported and doesn't provide these vars

This is not quite true, just use `ATTRS_SH_FILE` and `ATTRS_JSON_FILE` as in this removed snippet. Should be kept.

Reply:
@sternenseemann first of all sorry, I've seen that you were the author of that change and I intended to request a review from you, but I somehow forgot to add you to the list of reviewers. :facepalm: 

> just use ATTRS_SH_FILE and ATTRS_JSON_FILE as in this removed snippet

I added those in https://github.com/NixOS/nix/commit/3944a120ec6986c723bf36bfade9b331dd4af68a and AFAICS this never reached 2.3. More context is in https://github.com/NixOS/nixpkgs/pull/257919/commits/6884aba7363b1b4f2028a8a69976dd2cf55a5721.

Reply:
Right, seems like I got indeed fooled with the diff! The workaround code for `NIX_ATTRS_*_FILE` being empty seems to be in place still, so this should be unproblematic for Nix 2.3.